Code/Tools/ProjectManager/Source/CreateAGemScreen.cpp

```diff
@@ -0,0 +1,446 @@
+
+            m_license->setErrorLabelVisible(!licenseNameIsValid);
+        }
+        return licenseNameIsValid;
+    }
+
+    bool CreateAGemScreen::ValidateCreatorName()
+    {
+        bool creatorNameIsValid = true;
+        if (m_tabWidget->currentIndex() == 2)
+        {
+            m_numNextClicks += 1;
+            if (m_numNextClicks > 1)
+            {
+                if (m_creatorName->lineEdit()->text().isEmpty())
+                {
+                    creatorNameIsValid = false;
+                    m_creatorName->setErrorLabelText(tr("A creator's name is required."));
+                }
+
+                m_creatorName->setErrorLabelVisible(!creatorNameIsValid);
+            }
+        }
+        return creatorNameIsValid;
+    }
+
+    bool CreateAGemScreen::ValidateRepositoryURL()
+    {
+        bool repositoryURLIsValid = true;
+        if (m_tabWidget->currentIndex() == 2)
+        {
+            if (m_numNextClicks > 1)
+            {
+                if (m_repositoryURL->lineEdit()->text().isEmpty())
+                {
+                    repositoryURLIsValid = false;
+                    m_repositoryURL->setErrorLabelText(tr("A repository URL is required."));
+                }
+
+                m_repositoryURL->setErrorLabelVisible(!repositoryURLIsValid);
+            }
+        }
+        return repositoryURLIsValid;
+    }
+
+    void CreateAGemScreen::OnGemDisplayNameUpdated()
+    {
+        ValidateGemDisplayName();
+    }
+
+    void CreateAGemScreen::OnGemSystemNameUpdated()
+    {
+        ValidateGemSystemName();
+    }
+
+    void CreateAGemScreen::OnLicenseNameUpdated()
+    {
+        ValidateLicenseName();
+    }
+
+    void CreateAGemScreen::OnCreatorNameUpdated()
+    {
+        ValidateCreatorName();
+    }
+
+    void CreateAGemScreen::OnRepositoryURLUpdated()
+    {
+        ValidateRepositoryURL();
+    }
+
+    void CreateAGemScreen::HandleBackButton()
+    {
+        if (m_tabWidget->currentIndex() == 2)
+        {
+            m_tabWidget->setCurrentIndex(1);
+            m_nextButton->setText("Next");
+        }
+        else if (m_tabWidget->currentIndex() == 1)
+        {
+            m_tabWidget->setCurrentIndex(0);
+        }
+    }
+
+    void CreateAGemScreen::HandleNextButton()
+    {
+        if (m_tabWidget->currentIndex() == 0)
+        {
+            m_tabWidget->setCurrentIndex(1);
+        }
+        else if (m_tabWidget->currentIndex() == 1)
+        {
+            m_tabWidget->setCurrentIndex(2);
+            m_nextButton->setText("Create");
+        }
+    }
+
+    void CreateAGemScreen::ConvertNextButtonToCreate()
+    {
+        if (m_tabWidget->currentIndex() == 0)
+        {
+            m_nextButton->setText("Next");
+        }
+        else if (m_tabWidget->currentIndex() == 1)
+        {
+            m_nextButton->setText("Next");
+        }
+        else if (m_tabWidget->currentIndex() == 2)
+        {
+            m_nextButton->setText("Create");
+        }
```

Comment:
```suggestion
        if (m_tabWidget->currentIndex() == 2)
        {
            m_nextButton->setText(tr("Create"));
        }
        else
        {
            m_nextButton->setText(tr("Next"));
        }
```
simplify and use translation

Reply:
Fixed.